cpp/src/arrow/type.h

```diff
@@ -467,16 +486,20 @@ struct ARROW_EXPORT TimestampType : public FixedWidthType {
 
   int bit_width() const override { return static_cast<int>(sizeof(int64_t) * 8); }
 
-  TimeUnit unit;
-
   explicit TimestampType(TimeUnit unit = TimeUnit::MILLI)
       : FixedWidthType(Type::TIMESTAMP), unit(unit) {}
 
+  explicit TimestampType(const std::string& timezone, TimeUnit unit = TimeUnit::MILLI)
```

Comment:
Shouldn't `timezone` default to `"UTC"`? Also maybe it should be the 2nd parameter?